tuf/ngclient/_internal/trusted_metadata_set.py

```diff
@@ -0,0 +1,464 @@
+# Copyright the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""Trusted collection of client-side TUF Metadata
+
+TrustedMetadataSet keeps track of current valid set of metadata for the client,
+and handles almost every step of the "Detailed client workflow" (
+https://theupdateframework.github.io/specification/latest#detailed-client-workflow)
+in the TUF specification: the remaining steps are related to filesystem and
+network IO which is not handled here.
+
+Loaded metadata can be accessed via the index access with rolename as key
+(trusted_set["root"]) or, in the case of top-level metadata using the helper
+properties (trusted_set.root).
```

Comment:
Feel free to ignore this suggestion:
```suggestion
TrustedMetadataSet keeps track of the current valid set of metadata for the client,
and handles almost every step of the "Detailed client workflow" (
https://theupdateframework.github.io/specification/latest#detailed-client-workflow)
in the TUF specification: the remaining steps are related to filesystem and
network IO, which are not handled here.

Loaded metadata can be accessed via index access with rolename as key
(trusted_set["root"]) or, in the case of top-level metadata, using the helper
properties (trusted_set.root).
```